datashuttle/tui/tabs/transfer.py

```diff
@@ -9,5 +33,352 @@ def __init__(self, mainwindow, project):
+            for key in ["different", "local_only", "error"]
+        }
+        transfer_tree.diff_paths = [
+            path for category in filtered_diffs.values() for path in category
+        ]
+        transfer_tree.reload()
+
+
+class TransferStatusTree(FilteredTree):
+    def __init__(self, parent_tab, project, id=None):
+        super(TransferStatusTree, self).__init__(
+            project.get_local_path(), id=id
+        )
+
+        self.tab = parent_tab
+        self.project = project
+        self.transfer_diffs = get_local_and_central_file_differences(
+            project.cfg
+        )
+        filtered_diffs = {
+            key: self.transfer_diffs[key]
+            for key in ["different", "local_only", "error"]
+        }
+        self.diff_paths = [
+            path for category in filtered_diffs.values() for path in category
+        ]
+
+    def on_mount(self):
+        self.tab.transfer_paths = self.tab.get_transfer_paths()
+
+    def render_label(
+        self, node: TreeNode[DirEntry], base_style: Style, style: Style
+    ) -> Text:
+        node_label = node._label.copy()
+        node_label.stylize(style)
+
+        node_path = node.data.path
+
+        if node._allow_expand:
+            prefix = (
+                "📂 " if node.is_expanded else "📁 ",
+                base_style + TOGGLE_STYLE,
+            )
+            node_label.stylize_before(
+                self.get_component_rich_style(
+                    "directory-tree--folder", partial=True
+                )
+            )
+        else:
+            prefix = (
+                "📄 ",
+                base_style,
+            )
+            node_label.stylize_before(
+                self.get_component_rich_style(
+                    "directory-tree--file", partial=True
+                ),
+            )
+            node_label.highlight_regex(
+                r"\..+$",
+                self.get_component_rich_style(
+                    "directory-tree--extension", partial=True
+                ),
+            )
+
+        self.format_transfer_label(node, node_label, node_path)
+
+        if (
+            node_label.plain.startswith(".")
+            or node_path not in self.tab.transfer_paths
+        ):
+            node_label.stylize_before(
+                self.get_component_rich_style("directory-tree--hidden")
+                # "grey58"
+            )
+
+        text = Text.assemble(prefix, node_label)
+        return text
+
+    def format_transfer_label(self, node, node_label, node_path):
+        node_relative_path = node_path.as_posix().replace(
+            self.project.cfg.get_base_folder("local").as_posix() + "/", ""
+        )
+
+        if (
+            node_path.stem.startswith("sub-")
+            or node_path.stem.startswith("ses-")
+            or node_path.stem in get_datatypes()
+            # ) and not node.is_expanded:
+        ):
+            if any(node_relative_path in file for file in self.diff_paths):
```

Comment:
for readability I would consider assigning this boolean to a variable name e.g.:

```
path_has_changed = any(node_relative_path in file for file in self.diff_paths)  #(if I understood it's function correctly)

if path_has_changed:
   ...
```